DEVELOPMENT_GUIDE.md

```diff
@@ -328,6 +328,27 @@ conventions are best practices that we have learnt over time.
     comments.
     
     
+### Dependency Updates
+
+Please update all the required files if the changes involve a version update on a dependency or to include a new dependency. The requirements files are located inside the `requirements folder`.
+
+#### base.txt for SAM CLI code dependencies
+For dependencies used in SAM CLI code, update `base.txt` in `requirements` folder. To update `base.txt` file, simply follow the current convention and input the dependency name plus version, together with any necessary comment. For more information on the operators to be used for restricting compatible versions, read on [python's enhancement proposals](https://peps.python.org/pep-0440/#compatible-release).
+
+#### reproducible-linux.txt for SAM CLI code dependencies
+For dependencies used in SAM CLI code, also remember to update`reproducible-linux.txt` in `requirements` folder and `THIRD-PART-LICENSESD` in `installer/assets` folder. To update the `reproducible-linux.txt`, run the following script to replace the file:
+```
+make update-reproducible-reqs
+```
+Note that this is a fully auto-generated file, any manual changes to reproducible-linux.txt will not last after the next update running the above script. As for updating the `THIRD-PARTY-LICENSE`, find the corresponding dependency entry in the license file (usually grouped by licensing organization) and update the versions. For adding a new dependency, look up for its licensing organization through PyPi and update the corresponding section. If the license is from GNU or another license type not included in the file, please contact the repository maintainers first. If you are not familiar with working with this file, please contact one of the repository maintainers or cut an issue to help with the update.
+
+#### dev.txt for SAM CLI test dependencies
+For changing dependencies used for `make pr` checks and test related dependencies, update `dev.txt` in `requirements` folder only.
+
+#### pyinstaller-build.txt for SAM CLI native installer build dependencies
+For changing dependencies needed to run PyInstaller builds (to run `build-mac.sh` or `build-linux.sh` in `installer/pyinstaller` folder), modify `pyinstaller-build.txt`.
```

Comment:
For changing python dependencies needed for creating builds through Pyinstaller? does that sound better?

Reply:
Sounds good, let me re-word this a bit!